modules/ximgproc/src/radon_transform.cpp

```diff
@@ -0,0 +1,421 @@
+
+        for(int x = xMin; x <= xMax; ++x) {
+          float y = a*x + b;
+          int yFloor = static_cast<int>(std::floor(y));
+          float yUp = y - yFloor;
+          float yLow = 1.0f - yUp;
+          y = (float)(yFloor);
+          y = static_cast<float>(std::max((int)(y), -n));
+          y = static_cast<float>(std::min((int)(y), n-2));
+
+          dst.at<float>(rhoMax - r, theta-minAngle) += (float)(yLow)*static_cast<float>(src.at<T>((int)(y) + n, x + m))
+                              + (float)(yUp)*static_cast<float>(src.at<T>((int)(y) + n + 1, x + m));
+        }
+      }
+    }
+
+    // Range: [136, 179]
+    else if( theta <= 179 ) {
+      for(int r = 1; r <= rhoMax; ++r) {
+        float rho = (float)(r - rc);
+        float b = rho/sinTh;
+
+        int yMax = std::min((int)(a*m + b), n-1),
+            yMin = std::max((int)(-a*m + b), -n);
+
+        for(int y = yMin; y <= yMax; ++y) {
+          float x = (y - b)/a;
+          int xFloor = static_cast<int>(std::floor(x));
+          float xUp = x - xFloor;
+          float xLow = 1.0f - xUp;
+          x = (float)(xFloor);
+          x = static_cast<float>(std::max((int)(x), -m));
+          x = static_cast<float>(std::min((int)(x), m-2));
+
+          dst.at<float>(rhoMax - r, theta-minAngle) += (float)(xLow)*static_cast<float>(src.at<T>(y + n, (int)(x) + m))
+                              + (float)(xUp)*static_cast<float>(src.at<T>(y + n, (int)(x) + m + 1));
+        }
+      }
+    }
+
+    // theta = 180 degrees | Integrating along a vertical line
+    else if( theta == 180 ) {
+      int rhoOffset = static_cast<int>((rhoMax - src.cols)/2);
+      for(int x = 1; x <= src.cols; ++x) {
+        int r = x + rhoOffset;
+        for(int y = 0; y < src.rows; ++y) {
+          dst.at<float>(r, 180-minAngle) += static_cast<float>(src.at<T>(y, x-1));
+        }
+      }
+    }
+}
+
+/**
+ * [setMinMaxAngle Sets the given minimum and maximum angles if they are smaller and higher than the current values respectively]
+ * @param currentMinAngle [Current minimum angle]
+ * @param currentMaxAngle [Current maximum angle]
+ * @param minAngle [Given minimum angle to set as current minimum angle, referenced]
+ * @param maxAngle [Given maximum angle to set as current maximum angle, referenced]
+ */
+static inline void setMinMaxAngle( int & currentMinAngle,
+                                   int & currentMaxAngle,
+                                   int minAngle,
+                                   int maxAngle)
+{
+  if (minAngle < currentMinAngle)
+  {
+    currentMinAngle = minAngle;
+  }
+  if (maxAngle > currentMaxAngle)
+  {
+    currentMaxAngle = maxAngle;
+  }
+}
+
+/**
+ * [getAngleRange Computes minAngle and maxAngle for given angleRange - AR_x_y]
+ * @param angleRange [Given flag for angle range - AR_x_y]
+ * @param minAngle   [minAngle, referenced]
+ * @param maxAngle   [maxAngle, referenced]
+ */
+
+static inline void getAngleRange( int angleRange,
```

Comment:
IMHO it would be better to replace `int angleRange` with `int angleMax, int angleMin` in `radonTransform(...)` interface to extend the set of use cases function could be used in. The implementation anyway prepare result degree by degree and therefore doesn't depend on particular angle range as input. Moreover it doesn't provide the result as a set so for `angleRange == AR_1_45 + AR_180` the result is same as for `angleRange == 63` which is confusing